src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/tracker/MetricsTracker.java

```diff
@@ -0,0 +1,33 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.tracker;
+
+/**
+ *  Class to track previous latency and failure metrics to calculate point in time metrics. These values are updated
+ *  everytime collector calls respective stats API.
+ */
+public class MetricsTracker {
```

Comment:
do you need this class or can you just maintain the previous stats object itself?